lib/request.js

```diff
@@ -1,41 +1,17 @@
 import core from "@actions/core";
 import { request } from "@octokit/request";
-import { ProxyAgent, fetch as undiciFetch } from "undici";
+import { EnvHttpProxyAgent } from "undici";
 
 const baseUrl = core.getInput("github-api-url").replace(/\/$/, "");
 
-// https://docs.github.com/actions/hosting-your-own-runners/managing-self-hosted-runners/using-a-proxy-server-with-self-hosted-runners
-const proxyUrl =
-  process.env.https_proxy ||
-  process.env.HTTPS_PROXY ||
-  process.env.http_proxy ||
-  process.env.HTTP_PROXY;
-
-/* c8 ignore start */
-// Native support for proxies in Undici is under consideration: https://github.com/nodejs/undici/issues/1650
-// Until then, we need to use a custom fetch function to add proxy support.
-const proxyFetch = (url, options) => {
-  const urlHost = new URL(url).hostname;
-  const noProxy = (process.env.no_proxy || process.env.NO_PROXY || "").split(
-    ","
-  );
-
-  if (!noProxy.includes(urlHost)) {
-    options = {
-      ...options,
-      dispatcher: new ProxyAgent(String(proxyUrl)),
-    };
-  }
-
-  return undiciFetch(url, options);
-};
-/* c8 ignore stop */
+const envHttpProxyAgent = new EnvHttpProxyAgent();
 
 export default request.defaults({
   headers: {
     "user-agent": "actions/create-github-app-token",
   },
   baseUrl,
-  /* c8 ignore next */
-  request: proxyUrl ? { fetch: proxyFetch } : {},
+  request: {
+    dispatcher: envHttpProxyAgent,
```

Comment:
I don't think this will do anything. There is. no `dispatch` option in `options.request`, see all implemented options in https://github.com/octokit/request.js#request

What we have to do is to implement a `fetch` wrapper

```js
function fetchWithProxyAgent(url, options) {
  return fetch(url, { ... options, dispatcher: envHttpProxyAgent })
}
```

And then use pass that custom fetch implementation as `options.request.fetch`

```suggestion
    fetch : fetchWithProxyAgent,
```